src/components/RadioButtonGroup/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import { classes, st } from './RadioButtonGroup.st.css';
+import { TPAComponentProps } from '../../types';
+import { RadioButton } from '../RadioButton';
+import { RadioButtonTheme } from '../RadioButton/RadioButton';
+
+export interface RadioButtonGroupProps extends TPAComponentProps {
+  label?: string | React.ReactNode;
+  children?: any;
+  layout?: RadioButtonGroupLayout;
+  error?: boolean;
+  errorText?: string;
+  disabled?: boolean;
+  'data-hook'?: string;
+  defaultValue?: string;
+  name: string;
+  withSpacing?: boolean;
+  theme?: RadioButtonTheme;
+  onChange?(value: string): void;
+}
+
+interface DefaultProps {
+  defaultValue: string;
+  layout?: RadioButtonGroupLayout;
+}
+
+interface State {
+  count: number;
+  checkedValue: string;
+}
+export enum RadioButtonGroupLayout {
+  Horizontal = 'horizontal',
+  Vertical = 'vertical',
+}
+
+/** radio button group */
+export class RadioButtonGroup extends React.Component<
+  RadioButtonGroupProps,
+  State
+> {
+  static displayName = 'RadioButtonGroup';
+  static defaultProps: DefaultProps = {
+    defaultValue: '',
+    layout: RadioButtonGroupLayout.Vertical,
+  };
+
+  state = { count: 0, checkedValue: this.props.defaultValue };
```

Comment:
We don't want an inner state in the component.
We want the component to be fully controlled.
You can see CheckboxGroup's implementation